### PR TITLE
drivers: spi: Fixing null-pointer dereference in stm32

### PR DESCRIPTION
**Pull request by a8961713**
fixes issue #21935 
Signed-off-by: David D <a8961713@gmail.com>

### CODE REVIEW

**Comment by tbursztyka** [comment]
> Signed-off-by: a8961713 a8961713@gmail.com

Can't you put your real name here?

Split the patch in 2. The test deserves its own patch.

For the commit title:
"drivers: spi: Fixing null-pointer dereference in stm32"

And the message: 

s/fixes issue/Fixes  

Please apply also the gitlink, and checkpatch findings.

**Comment by a8961713** [comment]
> > Signed-off-by: a8961713 [a8961713@gmail.com](mailto:a8961713@gmail.com)
> 
> Can't you put your real name here?
> 
> Split the patch in 2. The test deserves its own patch.
> 
> For the commit title:
> "drivers: spi: Fixing null-pointer dereference in stm32"
> 
> And the message:
> 
> s/fixes issue/Fixes
> 
> Please apply also the gitlink, and checkpatch findings.

Ok, I'll do it tomorrow.

File: drivers/spi/spi_ll_stm32.c [hunk 1]
```diff
@@ -96,14 +96,14 @@ static void spi_stm32_shift_m(SPI_TypeDef *spi, struct spi_stm32_data *data)
 #endif
```
Comment (tbursztyka) [inline drivers/spi/spi_ll_stm32.c]: You commit message needs to be fixed as well:

Fixes #21935

We enforce this format, once the patch is merged the issue it then automatically closed (some script looks up for this format)

Comment (a8961713) [reply drivers/spi/spi_ll_stm32.c]: > You commit message needs to be fixed as well:
> 
> Fixes #21935
> 
> We enforce this format, once the patch is merged the issue it then automatically closed (some script looks up for this format)

Done

**Review by tbursztyka — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Comment by sylvioalves** [comment]
@a8961713 @tbursztyka I know this is old but can you share a link or documentation where it states that nops are needed when TX buff is NULL? I could only find this info in Linux `spi.h` file.
`/*
		* According to documentation, when sending NULL tx buf -
		*  NOP frames should be sent on MOSI line
		*/
		`

**Pull request merged**